aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -40,12 +46,28 @@ macro_rules! convert_impl {
 }
 
 impl<'r> WriteOpConverter<'r> {
-    convert_impl!(convert_resource, get_resource_state_value_metadata);
-
     convert_impl!(convert_module, get_module_state_value_metadata);
 
     convert_impl!(convert_aggregator, get_aggregator_v1_state_value_metadata);
 
+    pub(crate) fn convert_resource(
+        &self,
+        state_key: &StateKey,
+        move_storage_op: MoveStorageOp<(Bytes, Option<MoveTypeLayout>)>,
+        legacy_creation_as_modification: bool,
+    ) -> Result<(WriteOp, Option<MoveTypeLayout>), VMStatus> {
+        let result = self.convert(
+            self.remote.get_resource_state_value_metadata(state_key),
+            move_storage_op.clone(),
```

Comment:
Let's avoid clone here?

Reply:
I changed it to pass &move_storage_op instead. But now, there are a bunch of clones in `convert` functions though.